### PR TITLE
[macOS] Implement helper to get GitHub package download URL

### PR DESCRIPTION
**Pull request by dibir-magomedsaygitov**
# Description
In scope of this PR we are implementing helper to get GitHub package download URL as it was done for Windows in scope of [this PR](https://github.com/actions/virtual-environments/pull/4645/files#:~:text=function%20Get%2DGitHubPackageDownloadUrl%20%7B).

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/provision/core/stack.sh [hunk 1]
```diff
@@ -2,24 +2,24 @@
 source ~/utils/utils.sh
 
 echo "Get the latest Stack version..."
-StackRelease=$(curl -H "Authorization: token $API_PAT" -s "https://api.github.com/repos/commercialhaskell/stack/releases/latest")
-DownloadUrl=$(echo $StackRelease | jq -r '.assets[].browser_download_url | select(contains("osx-x86_64.tar.gz"))' | head -n 1)
-StackVersion=$(echo $StackRelease | jq -r '.name' | cut -c2-)
-StackArchive="/tmp/stack.tar.gz"
+stackRelease=$(curl -s "https://api.github.com/repos/commercialhaskell/stack/releases/latest")
```
Comment (miketimofeev) [inline images/macos/provision/core/stack.sh:5]: Please use the API_PAT here

File: images/macos/provision/utils/utils.sh [hunk 2]
```diff
@@ -171,3 +171,28 @@ configure_user_tccdb () {
...
+    local FILTER=$3
+    local VERSION=$4
+    local API_PAT=$5
+    local SEARCH_IN_COUNT="100"
+
+    if [ $API_PAT ]; then
+        json=$(curl -H "Authorization: token $API_PAT" -s "https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/releases?per_page=${SEARCH_IN_COUNT}")
+    else
+        json=$(curl -s "https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/releases?per_page=${SEARCH_IN_COUNT}")
+    fi
```
Comment (miketimofeev) [inline images/macos/provision/utils/utils.sh:187]: Maybe get rid of `else`?
```suggestion
    if [ $API_PAT ]; then
        AuthString="-H 'Authorization: token $API_PAT'"
    fi

    json=$(curl $AuthString -s "https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/releases?per_page=${SEARCH_IN_COUNT}")
```

File: images/macos/provision/utils/utils.sh [hunk 3]
```diff
@@ -171,3 +171,26 @@ configure_user_tccdb () {
...
+
+get_github_package_download_url() {
+    local REPO_OWNER=$1
+    local REPO_NAME=$2
+    local FILTER=$3
+    local VERSION=$4
+    local API_PAT=$5
+    local SEARCH_IN_COUNT="100"
+
+    [ $API_PAT ] && authString="-H 'Authorization: token $API_PAT'"
```
Comment (al-cheb) [inline images/macos/provision/utils/utils.sh:183]: This condition doesn't work. I think we should check is string is not empty with -n flag.

**Review by al-cheb — APPROVED** [review]

**Comment by dibir-magomedsaygitov** [comment]
@miketimofeev for these 2 scripts releases has a different json structure and the helper can't be implemented for them.

**Review by miketimofeev — APPROVED** [review]

**Review by dmitry-shibanov — APPROVED** [review]

**Pull request merged**